lib/Controller/SettingsController.php

```diff
@@ -152,6 +155,22 @@ public function setSettings($wopi_url,
 		}
 
 		$this->discoveryManager->refretch();
+		try {
+			$capaUrlSrc = $this->wopiParser->getUrlSrc('Capabilities');
+			if (is_array($capaUrlSrc) && $capaUrlSrc['action'] === 'getinfo') {
+				// XXX Works, but probably better to strip trailing '/hosting/capabilities'
+				$public_wopi_url = $capaUrlSrc['urlsrc'];
```

Comment:
Since collabora is always using the hostname it was called with this will still be the private hostname that Nextcloud uses to call the discovery endpoint.